mmv1/third_party/terraform/services/developerconnect/resource_developer_connect_insights_config_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    Type: MMv1     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package developerconnect_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDeveloperConnectInsightsConfig_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDeveloperConnectInsightsConfig_basic(context),
+			},
+			{
+				ResourceName:            "google_developer_connect_insights_config.my_insights_config",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"insights_config_id", "labels", "location", "terraform_labels", "workload"},
+			},
+			{
+				Config: testAccDeveloperConnectInsightsConfig_update(context),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("google_developer_connect_insights_config.my_insights_config", plancheck.ResourceActionDestroyBeforeCreate),
+					},
+				},
+			},
+			{
+				ResourceName:            "google_developer_connect_insights_config.my_insights_config",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"insights_config_id", "location", "labels", "terraform_labels", "workload"},
+			},
+		},
+	})
+}
+
+func testAccDeveloperConnectInsightsConfig_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+	data "google_project" "project" {
+	  project_id = "devconnect-insights-terraform"
```

Comment:
We'll want the test to be as self-contained as possible.

* Are there bootstrapping steps performed in this project we can do as part of the test?
* If not, can we make them against the test environments (We support applying Terraform or running Go code for bootstrapping envs)?
* If we do need an externally managed project we'll need to discuss managing some things like service account (test runner) permissions offline.